app/models/concerns/account_counters.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module AccountCounters
+  extend ActiveSupport::Concern
+
+  included do
+    has_one :account_stat, inverse_of: :account
+    after_initialize :set_account_stat
+    after_save :save_account_stat
+  end
+
+  def statuses_count
+    account_stat&.statuses_count || 0
```

Comment:
should be `delegate :statuses_count, :statuses_count=, :following_count, :following_count=, :followers_count, :followers_count=, to: :account_stat`

and then handle the case of nil account_stat in the `set_account_stat` hook

Reply:
`%w(statuses following followers).flat_map { |i| [:"#{i}_count", :"#{i}_count=" }` is even a little cuter, your call.